### PR TITLE
cli: run JavaScript modules + provide `console` APIs

### PR DESCRIPTION
**Pull request by bajtos**
Start the implementation of Zinnia CLI, using `clap` for argument parsing.

Start the implementation of Zinnia Runtime, using `deno_core` as the JS/WASM engine.

Leverage useful bits from `deno_runtime`, like adding colours to terminal output and `console` Web-platform APIs.

As a general rule, I am trying to follow Deno as much as possible. Using the same conventions and coding style as they do, importing from their crates where possible and copying bits of code where it's difficult to import.

See https://github.com/filecoin-station/zinnia/issues/5

### CODE REVIEW

File: cli/main.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#[command(author, version, about, long_about = None)]
+pub struct CliArgs {
+  #[command(subcommand)]
+  command: Commands,
+}
+
+#[derive(Subcommand)]
+pub enum Commands {
+  Run {
+    /// JavaScript file containing the Station Module to run
```
Comment (patrickwoodhead) [inline cli/main.rs:24]: Three /// here, should be two?

**Review by patrickwoodhead — APPROVED** [review]

File: cli/main.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      println!("{} {file}", colors::green("EXECUTE"));
+      let main_module = deno_core::resolve_path(&file)?;
+      run_js_module(&main_module, &Default::default()).await?;
+      Ok(())
+    }
+  }
+}
+
+fn exit_with_error(error: AnyError) {
+  // Inspired by unwrap_or_exit<T> from https://github.com/denoland/deno/blob/main/cli/main.rs
```
Comment (juliangruber) [inline cli/main.rs:53]:
```suggestion
  // Inspired by unwrap_or_exit<T> from https://github.com/denoland/deno/blob/600fff79cdf5d52154344a0e3a8a523e1e21c3c1/cli/main.rs
```

Permalinks have higher chances or surviving

**Review by juliangruber — APPROVED** [review]
(with limited Rust knowledge) LGTM!

File: cli/main.rs (hunk 1 above)
Comment (bajtos) [reply cli/main.rs:24]: Great question!

This is a documentation comment, see https://doc.rust-lang.org/reference/comments.html#doc-comments

File: cli/main.rs (hunk 2 above)
Comment (bajtos) [reply cli/main.rs:53]: > Permalinks have higher chances or surviving

Yeah, I usually use permalinks too.

In this PR, I prefer to link to the latest version of the Deno source code to make it easier to find the latest version. That's why I am linking to the entire file, not only to the line range for this particular method.

Comment (juliangruber) [reply cli/main.rs:53]: Yes, but this entire file could be removed in the future! With a permalink you will always see at least what you saw when you wrote the code

Comment (bajtos) [reply cli/main.rs:53]: You are right. I reconsidered my original opinion and opened https://github.com/filecoin-station/zinnia/pull/16

**Pull request merged**